### PR TITLE
chore: Move changelog into tracker [DHIS2-17714]

### PR DESCRIPTION
**Pull request by enricocolasante**
## Big picture

Tracker has multiple services for each entity like tracked entity, enrollment, event and relationship. One is in dhis-api and one in dhis-service-tracker. Our goal is to provide one API (service) per entity that is part of the tracker domain/team.

Trackers architecture splits read/write into exporter services and an importer. So if you want to get data you use an exporter. If you want to insert, update or delete you have to go through the importer. Only then can we ensure integrity of tracker data.

Another goal is that code that provides tracker functionality and is thus owned by the tracker team lives in ideally one maven module (We can settle on a name later on maybe `dhis-service-tracker` or `dhis-tracker`).

This is a **big** task that we are going to implement in many small steps.

## This PR
- Move `TrackedEntityAttributeValueChangeLogService` and the store into tracker.
- Move `HibernateTrackedEntityDataValueChangeLogStore` into tracker.
- Delete `TrackedEntityAttributeDataValueChangeLogStore` interface as it is only used by the service in the same package.
- `TrackedEntityAttributeValueChangeLogStore` cannot be removed as it is used in `HibernatePotentialDuplicateStore`.
- Merge `TrackedEntityAttributeChangeLogService` and `TrackedEntityChangeLogService` in one service, since both have to do with change logs in tracked entities.
- Make `dhis-service-administration` module depend on `dhis-service-tracker` module.

### CODE REVIEW

**Review by teleivo — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/DefaultMaintenanceService.java [hunk 1]
```diff
@@ -171,7 +171,7 @@ public boolean pruneData(DataElement dataElement) {
       return false;
     }
 
-    manager.delete(dataElement);
```
Comment (ameenhere) [inline dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/DefaultMaintenanceService.java:174]: Curious about what this used to do, did it actually delete the metadata? 🤔

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityChangeLogService.java [hunk 2]
```diff
@@ -64,7 +65,57 @@ public class DefaultTrackedEntityChangeLogService implements TrackedEntityChange
...
+    // we do not have ACL API to check TE attributes.
+
+    Set<String> allUserReadableTrackedEntityAttributes =
+        trackedEntityAttributeService
+            .getAllUserReadableTrackedEntityAttributes(CurrentUserUtil.getCurrentUserDetails())
+            .stream()
+            .map(IdentifiableObject::getUid)
+            .collect(Collectors.toSet());
+
+    return attributeValueChangeLogs.stream()
```
Comment (zubaira) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityChangeLogService.java:98]: what if `allUserReadableTrackedEntityAttributes` is empty? can we return early ?

File: dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/DefaultMaintenanceService.java (hunk 1 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/DefaultMaintenanceService.java:172]: This was wrongly introduced a few PRs ago
https://github.com/dhis2/dhis2-core/commit/420326b1da4b1bd0ff4da6a8e96afdf66a871ba7#diff-10acfa7c6114045775d7939221d3d89bb04ecbf87cff67d6e35a8ca91fba9c1f
We shouldn't delete the `dataElement` here, only the attached values

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityChangeLogService.java (hunk 2 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityChangeLogService.java:98]: Maybe.
I am just moving some code in this PR, I would prefer not to touch anything at the moment

**Review by zubaira — APPROVED** [review]
LGTM

**Pull request merged**